src/pykeen/training/training_loop.py

```diff
@@ -412,8 +464,23 @@ def _train(  # noqa: C901
                 'prev_loss': self.losses_per_epochs[-2] if epoch > 2 else float('nan'),
             })
 
+            # Save the last successful finished epoch
+            self._epoch = epoch
+
             if stopper is not None and stopper.should_evaluate(epoch) and stopper.should_stop(epoch):
+                # If a checkpoint file is given, we check whether it is time to save a checkpoint
+                if checkpoint_file:
+                    minutes_since_last_checkpoint = (time.time() - last_checkpoint) // 60
+                    if minutes_since_last_checkpoint >= checkpoint_frequency:
+                        self._save_state(path=checkpoint_file, stopper=stopper)
                 return self.losses_per_epochs
+            else:
+                # If a checkpoint file is given, we check whether it is time to save a checkpoint
+                if checkpoint_file:
+                    minutes_since_last_checkpoint = (time.time() - last_checkpoint) // 60
+                    if minutes_since_last_checkpoint >= checkpoint_frequency:
```

Comment:
there's some duplication of code here, can it be reduced?

Reply:
I considered creating a function, but that function would need the `checkpoint_file`, `last_checkpoint` and `stopper` as argument. In addition, it would have to return the variable `last_checkpoint`, which would have to be assigned within the training_loop. Imo this would be a worse solution than it is right now.

Reply:
The only other possibility I came up with is
```
            should_stop = False
            if stopper is not None and stopper.should_evaluate(epoch) and stopper.should_stop(epoch):
                should_stop = True

            # If a checkpoint file is given, we check whether it is time to save a checkpoint
            if checkpoint_file:
                minutes_since_last_checkpoint = (time.time() - last_checkpoint) // 60
                if minutes_since_last_checkpoint >= checkpoint_frequency:
                    self._save_state(path=checkpoint_file, stopper=stopper)
                    last_checkpoint = time.time()
            
            if should_stop:
                return self.losses_per_epochs
```

Reply:
Which one do you prefer?

Reply:
I have added it now with https://github.com/pykeen/pykeen/pull/123/commits/7b84f70586c7e80f5277ffa8e26304da81f01825.